rfc/src/rfcs/0011-list.md

```diff
@@ -0,0 +1,157 @@
+This subcommand will not run verification; it will exit after printing the information.
+
+There will be two options:
+- `--filter=[harnesses|contracts]`: Only output the information specified. In the example above, `--filter harnesses` would output the `mod verify` entry and `--filter contracts` would output the `mod implementation` entry. In the case where harnesses and contracts are in the same module, Kani will output only the functions that are harnesses or have contracts, depending on the filter.
+- `--message-format=[human|json]`: Specify the output format. The default is `human`, which prints to the terminal. The `json` option creates and writes to a JSON file instead.
+
+This subcommand will not fail. In the case that it does not find any Kani attributes, it will print a message informing the user of that fact.
+
+### Output Format
+
+The default format, `human`, will print the package's modules as a tree. The leaves of the tree are functions with Kani attributes. If a module has no Kani attributes, Kani doesn't print it. After printing the tree, Kani will print the total count of harnesses, stubs, and contracts.
+For example:
+
+```
+crate example
+├── mod implementation
+│   └── fn foo 
+|       - requires-contracts:
+|          - ub_checks::can_dereference(x)
+|          - ub_checks::can_write(x)
+|       - ensures-contracts: 
+|           - |result| *result == *x + 1
+|       - modifies-contracts: 
+|           - x
+|       - proofs-for-contracts: 
+|           - verify::check_foo_u32, 
+|           - verify::check_foo_u64
+├── mod verify
+|   └── fn check_foo_u32
+|       - kani::proof_for_contract(foo)
+|   └── fn check_foo_u64
+|       - kani::proof_for_contract(foo)
+│   └── fn check_func
+|       - kani::proof
+|       - kani::should_panic
+|       - kani::stub(rand::random, mock_random)
+|   └── fn check_bar
+|       - kani::proof
+|       - kani::unwind(5)
+
+Totals:
+    - Harnesses: 4
+    - Requires Contracts: 2
+    - Ensures Contracts: 1
+    - Modifies Contracts: 1
+    - Stubs: 1
+```
+
+As the name implies, the goal of the `human` output is to be friendly for human readers. If the user wants an output format that's more easily parsed by a script, they can opt for the `json` option, which will output the same information in a more verbose, machine-readable format. By "more verbose" we mean that it will parse the attributes, so that instead of outputting `kani::stub(rand::random, mock_random)`, it would output:
+
+```json
+stub : {
+    original: rand::random,
+    target: mock_random
+}
+```
+and so on for other attributes.
+
+## Software Design
+
+We will add a new subcommand to `kani-driver` which follows the normal verification workflow of invoking `kani-compiler` and constructing a `Project` from the result. Then, instead of verifying the project, Kani will iterate over the `KaniMetadata` in the `Project` to construct the output.
+
+### Contracts
```

Comment:
Is this really necessary? I would prefer if we keep it simple, and leave this as an open question for now. I'm not sure we want the `list` sub-command to print all the attributes as they are.